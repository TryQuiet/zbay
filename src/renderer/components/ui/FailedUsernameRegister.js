import React from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'

import red from '@material-ui/core/colors/red'

import Icon from './Icon'
import exclamationMark from '../../static/images/exclamationMark.svg'
import Modal from './Modal'
import LoadingButton from './LoadingButton'

const styles = theme => ({
  root: {
    padding: theme.spacing(4)
  },
  icon: {
    fontSize: '10rem',
    color: red[500],
    width: 80,
    height: 70
  },
  message: {
    wordBreak: 'break-all',
    marginTop: 36,
    fontWeight: 500
  },
  info: {
    textAlign: 'center'
  },
  button: {
    textTransform: 'none',
    width: 187,
    height: 60,
    fontSize: 16,
    lineHeight: '19px',
    fontWeight: 'normal',
    marginTop: 24
  }
})

export const FailedUsernameRegister = ({
  classes,
  open,
  handleClose,
  openModalCreateUsername
}) => {
  return (
    <Modal open={open} handleClose={handleClose} title=''>
      <Grid
        container
        justify='flex-start'
        spacing={3}
        direction='column'
        className={classes.root}
      >
        <Grid item container direction='column' alignItems='center'>
          <Icon className={classes.icon} src={exclamationMark} />
          <Typography variant='h3' className={classes.message}>
            Oops!
          </Typography>
        </Grid>
        <Grid item container direction='column'>
          <Grid item>
            <Typography variant='body2' className={classes.info}>
              Username didn't get registered. You are still anonymous.
            </Typography>
          </Grid>
          <Grid item container justify='center' alignItems='center'>
            <LoadingButton
              classes={{ button: classes.button }}
              text='Create a username'
              onClick={() => {
                openModalCreateUsername()
                handleClose()
              }}
            />
          </Grid>
        </Grid>
      </Grid>
    </Modal>
  )
}

FailedUsernameRegister.propTypes = {
  classes: PropTypes.object.isRequired,
  open: PropTypes.bool.isRequired,
  handleClose: PropTypes.func.isRequired,
  openModalCreateUsername: PropTypes.func.isRequired
}

FailedUsernameRegister.defaultProps = {
  open: false
}

export default R.compose(React.memo, withStyles(styles))(FailedUsernameRegister)
