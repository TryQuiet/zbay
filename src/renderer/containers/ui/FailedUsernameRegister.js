import React from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import * as R from 'ramda'

import FailedUsernameRegister from '../../components/ui/FailedUsernameRegister'
import modalsHandlers, { withModal } from '../../store/handlers/modals'

export const mapStateToProps = state => ({})

export const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      openModalCreateUsername: modalsHandlers.actionCreators.openModal('createUsernameModal')
    },

    dispatch
  )

export default R.compose(
  React.memo,
  connect(mapStateToProps, mapDispatchToProps),
  withModal('failedUsernameRegister')
)(FailedUsernameRegister)
