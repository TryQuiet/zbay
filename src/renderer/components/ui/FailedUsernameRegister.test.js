/* eslint import/first: 0 */
import React from 'react'
import { shallow } from 'enzyme'

import { FailedUsernameRegister } from './FailedUsernameRegister'
import { mockClasses } from '../../../shared/testing/mocks'

describe('FailedUsernameRegister', () => {
  it('renders component', () => {
    const result = shallow(
      <FailedUsernameRegister
        open
        classes={mockClasses}
        handleClose={jest.fn()}
        openModalCreateUsername={jest.fn()}
      />
    )
    expect(result).toMatchSnapshot()
  })
})
