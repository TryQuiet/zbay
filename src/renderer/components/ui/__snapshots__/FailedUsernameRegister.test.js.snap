// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FailedUsernameRegister renders component 1`] = `
<Memo(WithStyles(Modal))
  handleClose={[MockFunction]}
  open={true}
  title=""
>
  <WithStyles(ForwardRef(Grid))
    className="root-mock"
    container={true}
    direction="column"
    justify="flex-start"
    spacing={3}
  >
    <WithStyles(ForwardRef(Grid))
      alignItems="center"
      container={true}
      direction="column"
      item={true}
    >
      <Memo(Icon)
        className="icon-mock"
        src="test-file-stub"
      />
      <WithStyles(ForwardRef(Typography))
        className="message-mock"
        variant="h3"
      >
        Oops!
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      container={true}
      direction="column"
      item={true}
    >
      <WithStyles(ForwardRef(Grid))
        item={true}
      >
        <WithStyles(ForwardRef(Typography))
          className="info-mock"
          variant="body2"
        >
          Username didn't get registered. You are still anonymous.
        </WithStyles(ForwardRef(Typography))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(ForwardRef(Grid))
        alignItems="center"
        container={true}
        item={true}
        justify="center"
      >
        <WithStyles(LoadingButton)
          classes={
            Object {
              "button": "button-mock",
            }
          }
          inProgress={false}
          onClick={[Function]}
          text="Create a username"
        />
      </WithStyles(ForwardRef(Grid))>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</Memo(WithStyles(Modal))>
`;
